packages/compiler-cli/src/main.ts

```diff
@@ -56,14 +56,33 @@ export function main(
 }
 
 export function mainDiagnosticsForTest(
-    args: string[], config?: NgcParsedConfiguration): ReadonlyArray<ts.Diagnostic|api.Diagnostic> {
+    args: string[], config?: NgcParsedConfiguration,
+    programReuse?: {program: api.Program | undefined},
```

Comment:
Why is this wrapped into an object literal? It seems this can just be:

```suggestion
    programReuse?: api.Program,
```

Reply:
It's an input/output parameter. We're returning the new program back to the caller via the reference in this object. Not the best API; we'll overhaul it once VE no longer goes through it too.